endpoint/deploy.go

```diff
@@ -22,52 +28,49 @@ type ObjectNamer interface {
 }
 
 type DeployController interface {
-	Run(controller.DeployCmd) (interface{}, error)
+	Run(DeployCloudAppPayload) (interface{}, error)
 }
 
-// DeployHandler handles deploying to OpenShift
-type DeployHandler struct {
+// Deploy handles deploying to OpenShift
+type Deploy struct {
 	logger           Logger
 	deployController controller.Deploy
 	namer            Namespacer
 	objectNamer      ObjectNamer
 }
 
-// EnvVar defines an environment variable
-type EnvVar struct {
-	Key   string `json:"key,omitempty"`
-	Value string `json:"value,omitempty"`
-}
-
 // DeployCloudAppPayload is the JSON structure sent from the client
 type DeployCloudAppPayload struct {
 	namer       Namespacer
 	objectNamer ObjectNamer
-	Domain      string    `json:"domain"`
-	EnvVars     []*EnvVar `json:"envVars,omitempty"`
-	Environment string    `json:"environment"`
-	GUID        string    `json:"guid"`
+	Domain      string               `json:"domain"`
+	EnvVars     []*controller.EnvVar `json:"envVars,omitempty"`
+	Environment string               `json:"environment"`
+	GUID        string               `json:"guid"`
+	Service     string               `json:"service"`
 	// this are the services you require such as redis
-	InfraServices []string `json:"infraServices,omitempty"`
-	Namespace     string   `json:"namespace"`
-	ProjectGUID   string   `json:"projectGuid"`
-	RepoURL       string   `json:"repoUrl,omitempty"`
-	RepoBranch    string   `json:"repoBranch,omitempty"`
-	User          string   `json:"user"`
-	Auth          string   `json:"auth"`
+	InfraServices      []string          `json:"infraServices,omitempty"`
+	Namespace          string            `json:"namespace"`
+	ProjectGUID        string            `json:"projectGuid"`
+	RepoURL            string            `json:"repoUrl,omitempty"`
+	RepoBranchOrCommit string            `json:"repoBranchOrCommit,omitempty"`
+	User               string            `json:"user"`
+	Auth               string            `json:"auth"`
+	AppTag             string            `json:"appTag"`
```

Comment:
this is along the lines of "cloud" or "redis" it's the bit that goes at the beginning of the generated name for the services and buildconfigs etc.